### PR TITLE
fix(core) message use histogram_quantile for max(histogram) queries.

### PR DESCRIPTION
**Pull request by yu-shipit**
max(histogram) is not supported histogram_quantile(1, histogram) may be what users want.

**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

**New behavior :**

**BREAKING CHANGES**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

Breaking changes may include:
- Any schema changes to any Cassandra tables
- The serialized format for Dataset and Column (see .toString methods)
- Over the wire formats for Akka messages / case classes
- Changes to the HTTP public API
- Changes to query parsing / PromQL parsing

**Other information**:

### CODE REVIEW

**Comment by yu-shipit** [comment]
max(histogram) will the the message below.
```
{
  "status" : "error",
  "data" : null,
  "errorType" : "requirement_failed",
  "error" : "max(histogram expr) is not supported. You may use histogram_quantile(1, histogram expr) instead.",
  "queryStats" : [ {
    "group" : [ "local", "raw", "aci-telemetry", "Test-data-1679598622_p1", "my_histogram" ],
    "timeSeriesScanned" : 2,
    "dataBytesScanned" : 259,
    "resultBytes" : 0,
    "cpuNanos" : 0
  } ]
}
```

**Review by alextheimer — APPROVED** [review]
Thanks @yu-shipit for digging into this unexpectedly-complex problem 👍

**Comment by amolnayak311** [comment]
1. If the behavior differs from Prometheus we need to call it out explicitly, in the PR and also a comment in the code as we generally strive to maintain compatibility as far as possible. 
2. It is important for the user to understand that histogram_quantile will give the approximation  which is dependent on their buckets, while the user might be expecting to see the max value recorded.

File: query/src/main/scala/filodb/query/exec/aggregator/RowAggregator.scala [hunk 1]
```diff
@@ -121,8 +121,8 @@ object RowAggregator {
   def apply(aggrOp: AggregationOperator, params: Seq[Any], schema: ResultSchema): RowAggregator = {
     val valColType = ResultSchema.valueColumnType(schema)
     aggrOp match {
-      case Min      => MinRowAggregator
-      case Max      => MaxRowAggregator
+      case Min if valColType == ColumnType.DoubleColumn => MinRowAggregator
```
Comment (amolnayak311) [inline query/src/main/scala/filodb/query/exec/aggregator/RowAggregator.scala:124]: Do you see any case where `LongColumn` and `IntColumn` would need this? I understand we don't want to use this for histograms but by restricting to DoubleColumn only we also don't support Long and Int columns

**Review by amolnayak311 — APPROVED** [review]

File: query/src/main/scala/filodb/query/exec/aggregator/RowAggregator.scala [hunk 2]
```diff
@@ -121,22 +121,26 @@ object RowAggregator {
...
+      case BottomK if valColType == ColumnType.DoubleColumn
+                  => new TopBottomKRowAggregator(params(0).asInstanceOf[Double].toInt, true)
+      case Quantile if valColType == ColumnType.DoubleColumn
+          => new QuantileRowAggregator(params(0).asInstanceOf[Double])
+      case Stdvar if valColType == ColumnType.DoubleColumn  => StdvarRowAggregator
+      case Stddev if valColType == ColumnType.DoubleColumn  => StddevRowAggregator
       case CountValues => new CountValuesRowAggregator(params(0).asInstanceOf[String])
-      case _     => ???
+      case _ => throw new IllegalArgumentException(s"This operation ${aggrOp} is not supported for ${valColType}." +
+      s" Because it does not result in a meaningful query. Please contact us if you have a use case.")
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/aggregator/RowAggregator.scala:143]: Let us not use this message as fallback for all.
For histograms:
> "The ${aggrOp} operation is not supported directly for histogram types. Check if you need to resolve to the individual histogram buckets, or calculate the rate and histogram_quantile before applying the aggregation. If you have a genuine use case for this query, please get in touch."


For doubles, it should be an illegal state exception. It should be unreachable code.

**Review by vishramachandran — CHANGES_REQUESTED** [review]
Asking for a small change in error message

File: query/src/main/scala/filodb/query/exec/aggregator/RowAggregator.scala (hunk 2 above)
Comment (vishramachandran) [reply query/src/main/scala/filodb/query/exec/aggregator/RowAggregator.scala:143]: Count_values should also be applied only to doubles.

**Review by vishramachandran — APPROVED** [review]
LGTM. thank you!!

**Pull request merged**